### PR TITLE
Document new shader built-ins (node position, camera pos/dir)

### PR DESCRIPTION
**Pull request by paddy-exe**
Following up on this PR: https://github.com/godotengine/godot/pull/63597
Documents the new shader built-ins introduced by this PR.
```
NODE_POSITION_WORLD = Node world space position
NODE_POSITION_VIEW = Node view position
CAMERA_POSITION_WORLD = Camera world space position
CAMERA_DIRECTION_WORLD = Camera world space direction
```

### CODE REVIEW

**Comment by clayjohn** [comment]
You can delete the examples you added before that overlap with these now

**Comment by Calinou** [comment]
> I generally agree but wouldn't it still make sense to keep them so people understand what happens "under the hood"?

I think the new constants can have a note of the form: "Equivalent to (whatever formula is used  behind the scenes)"

**Comment by clayjohn** [comment]
> I generally agree but wouldn't it still make sense to keep them so people understand what happens "under the hood"?

I don't think its helpful for users to have that detail in the docs if we don't expect them to use it. When writing documentation it is important to show the "right" way of doing things. If we show them the right way and a wrong way it will lead to confusion.

If this was a long-form tutorial and not a reference doc I would agree with you as long as we put a sufficient explanation as it would be a good learning opportunity. But since this is a reference doc we should keep it as streamlined as possible

File: tutorials/shaders/shader_reference/spatial_shader.rst [hunk 1]
```diff
@@ -229,28 +237,13 @@ The default use of a Godot fragment processor function is to set up the material
...
 | Built-in                               | Description                                                                                      |
 +========================================+==================================================================================================+
 | in vec2 **VIEWPORT_SIZE**              | Size of viewport (in pixels).                                                                    |
 +----------------------------------------+--------------------------------------------------------------------------------------------------+
-| in vec4 **FRAGCOORD**                  | Coordinate of pixel center in screen space. ``xy`` specifies position in window, ``z``           |
-|                                        | specifies fragment depth if ``DEPTH`` is not used. Origin is lower-left.                         |
+|| in vec4 **FRAGCOORD**                 || Coordinate of pixel center in screen space. ``xy`` specifies position in window, ``z``          |
+||                                       || specifies fragment depth if ``DEPTH`` is not used. Origin is lower-left.                        |
```
Comment (clayjohn) [inline tutorials/shaders/shader_reference/spatial_shader.rst:246]: I think there are a few too many "|" here

File: tutorials/shaders/shader_reference/spatial_shader.rst [hunk 2]
```diff
@@ -162,8 +162,16 @@ shader, this value can be used as desired.
...
+| in vec3 **NODE_POSITION_WORLD**        | Node world space position                              |
++----------------------------------------+--------------------------------------------------------+
+| in vec3 **NODE_POSITION_VIEW**         | Node view space position                               |
++----------------------------------------+--------------------------------------------------------+
+| in vec3 **CAMERA_POSITION_WORLD**      | Camera world space position                            |
++----------------------------------------+--------------------------------------------------------+
+| in vec3 **CAMERA_DIRECTION_WORLD**     | Camera world space direction                           |
++----------------------------------------+--------------------------------------------------------+
+|| in bool **OUTPUT_IS_SRGB**            || ``true`` when calculations happen in sRGB color space |
+||                                       || (``true`` in GLES2, ``false`` in GLES3).              |
```
Comment (mhilbrunner) [inline tutorials/shaders/shader_reference/spatial_shader.rst:174]:
```suggestion
|| in bool **OUTPUT_IS_SRGB**            || ``true`` when calculations happen in sRGB color space |
||                                       || (``true`` in GLES2, ``false`` in GLES3).              |
```
Same issue :)

Comment (mhilbrunner) [inline tutorials/shaders/shader_reference/spatial_shader.rst:174]: Same issue :)

File: tutorials/shaders/shader_reference/spatial_shader.rst [hunk 3]
```diff
@@ -278,6 +271,14 @@ Below are examples of common variables calculated using the built-ins:
...
+| in vec3 **NODE_POSITION_WORLD**        | Node world space position                                                                        |
++----------------------------------------+--------------------------------------------------------------------------------------------------+
+| in vec3 **NODE_POSITION_VIEW**         | Node view space position                                                                         |
++----------------------------------------+--------------------------------------------------------------------------------------------------+
+| in vec3 **CAMERA_POSITION_WORLD**      | Camera world space position                                                                      |
++----------------------------------------+--------------------------------------------------------------------------------------------------+
+| in vec3 **CAMERA_DIRECTION_WORLD**     | Camera world space direction                                                                     |                                                                                                  |
```
Comment (mhilbrunner) [inline tutorials/shaders/shader_reference/spatial_shader.rst:280]:
```suggestion
| in vec3 **CAMERA_DIRECTION_WORLD**     | Camera world space direction                                                                     |
```
There is stray whitespace and a pipe on line 280, which makes the table malformed. Everything else looks good!

Comment (paddy-exe) [reply tutorials/shaders/shader_reference/spatial_shader.rst]: Okay I think I resolved it. It seems to render correctly now :D

**Comment by mhilbrunner** [comment]
@paddy-exe Looking good now :) Last minor nitpick: can you add a '.' to the end of the descriptions for consistency with the existing ones? Then I'll merge it, promise :P Sorry I missed that before. Good work!

**Comment by paddy-exe** [comment]
@mhilbrunner Done, I think :D And no problem. My bad for not noticing it. Thanks for the quick feedback!

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks for the quick turnaround, and thank you for contributing to Godot's docs, it's appreciated! ❤️🎉

**Pull request merged**